src/application/components/MemoryInternals.tsx

```diff
@@ -0,0 +1,295 @@
+
+  const { data, networkStatus, error } = useQuery(MEMORY_INTERNALS_QUERY, {
+    variables: { clientId: clientId as string },
+    skip: !clientId,
+    pollInterval: 1000,
+  });
+
+  if (error) {
+    throw error;
+  }
+
+  const memoryInternals = data?.client?.memoryInternals;
+  const caches = memoryInternals?.caches;
+  const version = data?.client?.version;
+
+  if (networkStatus === NetworkStatus.loading) {
+    return (
+      <EmptyLayout>
+        <PageSpinner />
+      </EmptyLayout>
+    );
+  }
+
+  if (version && lt(version, "3.9.0")) {
+    return (
+      <EmptyLayout>
+        <PageError className="mt-8">
+          <PageError.Content>
+            <PageError.Title>Memory management not available</PageError.Title>
+            <PageError.Body>
+              Memory management is available in Apollo Client{" "}
+              <ExternalLink
+                href="https://github.com/apollographql/apollo-client/releases/tag/v3.9.0"
+                className="font-medium inline-flex items-center gap-2"
+              >
+                3.9.0
+                <IconOutlink className="size-4" />
+              </ExternalLink>{" "}
+              or greater. Please upgrade your Apollo Client version to use this
+              feature.
+            </PageError.Body>
+          </PageError.Content>
+        </PageError>
+      </EmptyLayout>
+    );
+  }
+
+  if (!caches) {
+    return (
+      <EmptyLayout>
+        <PageError className="mt-8">
+          <PageError.Content>
+            <PageError.Title>Unable to get memory internals</PageError.Title>
+            <PageError.Body>
+              Could not get memory internals for the client. This could be a
+              result of running your application in production mode since access
+              to memory internals is only available in development builds.
+            </PageError.Body>
+          </PageError.Content>
+        </PageError>
+      </EmptyLayout>
+    );
+  }
+
+  return (
+    <FullWidthLayout className="p-4 gap-6">
+      <div className="flex gap-2 justify-between items-start">
+        <header className="flex flex-col gap-2">
+          <h1 className="font-medium text-2xl text-heading dark:text-heading-dark">
+            Memory
+          </h1>
+          <p className="text-secondary dark:text-secondary-dark">
+            Learn how Apollo Client manages memory and how to set custom cache
+            size limits in the{" "}
+            <a
+              href="https://www.apollographql.com/docs/react/caching/memory-management/"
+              className="font-medium underline inline-flex items-center gap-2"
+              target="_blank"
+              rel="noopener noreferer noreferrer"
+            >
+              docs
+              <IconOutlink className="size-4" />
+            </a>
+            .
+          </p>
+        </header>
+
+        <ButtonGroup>
+          <Tooltip content="View historical">
```

Comment:
I'm not sure - is "historical" the right description here, or should it be something like "as dashboard"?